### DISCUSSION TITLE
Are paths pre-normalized?

### DISCUSSION THREAD

**Discussion by Dwedit** [dokan-dev/dokany#1010] [Q&A] [answered]
On Windows, the directory `".."` will erase the previous appearing directory in the path.  For example, these are valid paths on Windows:

`"C:\_INVALID_/.\_INVALID_/\/\/..\..\windows"`
`"C:\con\con\..\..\windows"`

Meanwhile on Linux, each piece of a path must exist.  Having ".." in a path does not remove a previously appearing piece of that path.  So `"/_INVALID_/../usr/bin"` is not a valid path on Linux.

Now my question is if Dokan will ever send a non-normalized path to the filesystem program?  I mean a path containing `".."` or `"."` as path pieces, multiple consecutive slashes, or mixed forwards slashes and backslashes.

**Comment by Liryna** [comment, marked as answer]
Hi @Dwedit ,

Dokan will always provide a normalized path. If you have a case where it is not, please let us know by opening an issue.
